pkg/controllers/disruption/helpers.go

```diff
@@ -157,15 +157,61 @@ func GetPodEvictionCost(ctx context.Context, p *v1.Pod) float64 {
 	return clamp(-10.0, cost, 10.0)
 }
 
-func filterByPrice(options []*cloudprovider.InstanceType, reqs scheduling.Requirements, price float64) []*cloudprovider.InstanceType {
+// FilterByPriceAndFindIncompatibleRequirementWithMinValues returns the instanceTypes that are lower priced than the current candidate and iterates over the cumulative minimum requirement of the InstanceTypeOptions to see if it meets the minValues of requirements.
+// The minValues requirement is checked again after filterByPrice as it may result in more constrained InstanceTypeOptions for a NodeClaim
+func FilterByPriceAndFindIncompatibleRequirementWithMinValues(options []*cloudprovider.InstanceType, reqs scheduling.Requirements, price float64) ([]*cloudprovider.InstanceType, string) {
```

Comment:
Do we take the same stance that we did with the other filter function where we say that the interface says that we don't return instance types unless we meet the minValues requirements?

Reply:
Yes, changed this behavior to return only when minValues requirement is met.